plugins/UsersManager/TokenNotifications/TokenProviderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UsersManager\TokenNotifications;
+
+interface TokenProviderInterface
+{
+    /**
+     * Provides a list of tokens to be notified, each with their information
+     * that can be used to populate the notification email
+     *
+     * @return TokenNotificationInterface[]
+     */
+    public function getTokensToNotify(): array;
```

Comment:
With the wording "getTokens" I would expect a pure data container, and not an object that has a "dispatch" functionality :thinking: 

Maybe something like `getPendingTokenNotifications`?